File: spring/src/main/java/cardsproject/domain/cards/DeckSideboardCard.java
package cardsproject.domain.cards;

import jakarta.persistence.*;

@Entity
@Table(name = "deck_sideboard_cards")
public class DeckSideboardCard {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Integer quantity = 1;

    @Column(name = "deck_id")
    private Long deckId;
    @Column(name = "card_id")
    private Long cardId;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Integer getQuantity() { return quantity; }
    public void setQuantity(Integer quantity) { this.quantity = quantity; }
    public Long getDeckId() { return deckId; }
    public void setDeckId(Long deckId) { this.deckId = deckId; }
    public Long getCardId() { return cardId; }
    public void setCardId(Long cardId) { this.cardId = cardId; }
}

File: symfony/src/Entity/Cards/DeckSideboardCard.php
<?php

namespace App\Entity\Cards;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;
use App\Repository\Cards\DeckSideboardCardRepository;

#[ORM\Entity(repositoryClass: DeckSideboardCardRepository::class)]
#[ORM\Table(name: 'deck_sideboard_card')]
class DeckSideboardCard
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    #[Groups(['deckSideboardCard:read'])]
    private ?int $id = null;

    #[ORM\Column(type: 'integer')]
    #[Groups(['deckSideboardCard:read', 'deckSideboardCard:write'])]
    private int $quantity = 1;

    #[ORM\ManyToOne(targetEntity: Deck::class, inversedBy: 'sideboard_cards')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Deck $deck = null;

    #[ORM\ManyToOne(targetEntity: Card::class, inversedBy: 'sideboard_decks')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Card $card = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getQuantity(): int
    {
        return $this->quantity;
    }

    public function setQuantity(int $quantity): static
    {
        $this->quantity = $quantity;
        return $this;
    }

    #[Groups(['deckSideboardCard:read'])]
    public function getDeckId(): ?int
    {
        return $this->deck?->getId();
    }

    public function getDeck(): ?Deck
    {
        return $this->deck;
    }

    public function setDeck(?Deck $deck): static
    {
        $this->deck = $deck;
        return $this;
    }

    #[Groups(['deckSideboardCard:read'])]
    public function getCardId(): ?int
    {
        return $this->card?->getId();
    }

    public function getCard(): ?Card
    {
        return $this->card;
    }

    public function setCard(?Card $card): static
    {
        $this->card = $card;
        return $this;
    }

    // ── Business operations ──────────────────────────────────────────

    public function increment($amount): void
    {
        throw new \RuntimeException('increment not implemented');
    }

    public function decrement($amount): void
    {
        throw new \RuntimeException('decrement not implemented');
    }

}
